test/services/fileStorage/index.test.js

```diff
@@ -404,6 +405,27 @@ describe('fileStorage services', () => {
 
 			return Promise.all(promises);
 		});
+
+		it('allows to override the internal file name with a request specific one', async () => {
+			const context = setContext('0000d224816abba584714c8e');
+			const requestedName = encodeURIComponent('some name-12345678id.png');
+			const spy = sinon.spy(AWSStrategy.prototype, 'getSignedUrl');
+
+			await signedUrlService
+				.find({
+					query: {
+						file: '5ca613c4c7f5120b8c5bef33',
+						name: requestedName,
+					},
+					...context,
+				})
+				.then(() => {
+					expect(spy.calledWithMatch({ localFileName: requestedName })).to.equal(true);
+				})
+				.finally(() => {
+					spy.restore();
+				});
```

Comment:
Just wondering about the mix of `async/await` and `Promise.prototype.then()/.finally()`. 🤔 

I'd have expected the following:

```suggestion
			try {
				await signedUrlService.find({
					query: {
						file: '5ca613c4c7f5120b8c5bef33',
						name: requestedName,
					},
					...context,
				});
				expect(spy.calledWithMatch({ localFileName: requestedName })).to.equal(true);
			} finally {
				spy.restore();
			}
```

🤷‍♂️

Reply:
@lwassermann do you want to change this?